cypress/integration/emoji/recently_used_emoji_spec.js

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [number] indicates a test step (e.g. 1. Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+/* eslint max-nested-callbacks: ["error", 3] */
+
+import {getRandomInt} from '../../utils';
+
+describe('Recent Emoji', () => {
+    before(() => {
+        // 1. Login and go to /
+        cy.login('user-1');
+        cy.visit('/');
+    });
+
+    it('M14014 Recently used emojis are shown 1st', async () => {
+        const maxRandom = 400;
+
+        // 2. Get random emoji index
+        const firstEmoji = getRandomInt(maxRandom);
+
+        // 3. Show emoji list
+        cy.get('#emojiPickerButton').should('be.visible').click();
+
+        // 4. Click emoji with random index
+        cy.get('.emoji-picker__item').eq(firstEmoji).click();
+
+        // 5. Submit post
+        cy.get('#create_post').submit();
+
+        // 6. Wait 500 millisecond
+        cy.wait(500); // eslint-disable-line
+
+        // 7. Post reaction to post
+        cy.clickPostReactionIcon();
+
+        // 8. Get second emoji
+        const secondEmoji = getRandomInt(maxRandom);
+
+        // 9. Click chosen emoji
+        cy.get('.emoji-picker__item').eq(secondEmoji).click();
+
+        // 10. Show emoji list
+        cy.get('#emojiPickerButton').click();
+
+        // * Assert first emoji should equal with second recent emoji
+        cy.get('.emoji-picker__item').eq(firstEmoji).find('img').then(($el) => {
```

Comment:
This is failing the test since `firstEmoji` and `secondEmoji` are moving targets.  Given that `firstEmoji` and `secondEmoji` are always unique, it should be,
```suggestion
        cy.get('.emoji-picker__item').eq(firstEmoji + 2).find('img').then(($el) => {
```

Reply:
Hi @saturninoabril,
After I hard coded `seconEmoji` and `firtsEmoji`, I don't get failed.
Is any testing steps that I didn't do yet?

FYI I just update to hard code `secondEmoji` and `firstEmoji`.

Reply:
@adzimzf Thank you for the update.
I think the test steps are perfectly fine but when I ran your updated commit, I got an error since it's validating against the different emoji.
![Screen Shot 2019-04-17 at 3 15 28 PM](https://user-images.githubusercontent.com/5334504/56268531-22148d80-6124-11e9-9b27-3579a4e3409b.png)
But after adjusting the values like `eq(firstEmoji + 2)` and `eq(secondEmoji + 1)`, I finally get it working.
![Screen Shot 2019-04-17 at 3 17 21 PM](https://user-images.githubusercontent.com/5334504/56268619-60aa4800-6124-11e9-87fa-7f0c2b0aba72.png)

Reply:
Hi @saturninoabril,
sorry for delay.
the problem is because when the length of recent emoji already reached maximal (27 emojis) it works use my last commit.
but when the length of recent emoji is less then maximal, it will fail. because the index of emoji is different between before and after adding recent emoji.

my approach is store current length of recent emoji, then when it assert will check if length is reach maximum it'll use `eq(firstEmoji)` and `eq(secondEmoji)` but if the length is less then maximum it'll use `eq(firstEmoji + 2)` and `eq(secondEmoji + 1)`.

but I got problem how to get length of recent emoji then store to a variable for used in assert.

Reply:
Sounds like a good plan.  We're storing recent emojis via localStorage with a key of `recent_emojis:{user_id}`.
<img width="756" alt="Screen Shot 2019-04-25 at 5 18 41 PM" src="https://user-images.githubusercontent.com/5334504/56725106-d21a7400-677e-11e9-83fd-23853f0c7d4c.png">
If you plan to get its length, you'll have to get emoji names first as `localStorage.getItem(recent_emojis:{user_id})` then count it.  You need to get the `user_id` as `cy.getCookie('MMUSERID').as('userId');`

Another approach is not to worry about the recent emojis at all by initially clearing the localStorage, like `cy.clearLocalStorage(/recent_emojis/)`

Reply:
I think your approach is clearer and easier to read.